plugins/catalog/src/components/Router.tsx

```diff
@@ -47,7 +48,7 @@ const DefaultEntityPage = () => (
 const EntityPageSwitch = ({ EntityPage }: { EntityPage: ComponentType }) => {
   const { entity } = useEntity();
   // Loading and error states
-  if (!entity) return <EntityPageLayout />;
+  if (!entity) return <EntityNotFound />;
```

Comment:
Oh no, that's probably not gonna work nicely. Now there is going to be a flickering when entity is being loaded, so the `EntityNotFound` will be first to show, then immediately after we get an entity it will be replaced with an `EntityPageLayout`.
You need to decouple loading from the error state, if loading then just return `<EntityPageLayout/>`.
https://github.com/spotify/backstage/blob/0d33fdba1e278da69c1b81bae4d4d433ac19377b/plugins/catalog/src/hooks/useEntity.ts#L70
You can also make `useEntity` hook to return if the entity is loading/error. (keep in mind that there is also case where the entity hasn't been fetched yet, hence `loading: false, entity: undefined, error: undefined`

Reply:
@shmidt-i I've made the changes.  Please check it

Reply:
LGTM, now some type fixes and can merge ! 😃

Reply:
Thank you. Where should I make the changes?